nova/virt/vmwareapi/volumeops.py

```diff
@@ -27,12 +27,25 @@
 from nova.i18n import _
 from nova.virt import driver
 from nova.virt.vmwareapi import constants
+from nova.virt.vmwareapi.session import StableMoRefProxy
 from nova.virt.vmwareapi import vm_util
 
 CONF = nova.conf.CONF
 LOG = logging.getLogger(__name__)
 
 
+class VolumeMoRefProxy(StableMoRefProxy):
+    def __init__(self, connection_info_data):
+        volume_ref_name = connection_info_data['volume']
+        ref = vutil.get_moref(volume_ref_name, 'VirtualMachine')
+        super(VolumeMoRefProxy, self).__init__(ref)
+        self._connection_info_data = connection_info_data
+
+    def fetch_moref(self, session):
+        volume_id = self._connection_info_data['volume_id']
+        self.moref = vm_util._get_vm_ref_from_vm_uuid(session, volume_id)
+
```

Comment:
Why don't we need `__eq__` here, but needed it in the stable vm ref PR?

Reply:
We do not really compare references directly in our code, except in the tests.
The proper references can only compared by their value. I fixed it for the stable vm too.